shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -185,15 +193,269 @@ - (void)testStartKeyboardAnimationWillInvokeSetupKeyboardAnimationVsyncClient {
+                    @"UIKeyboardAnimationDurationUserInfoKey" : [NSNumber numberWithDouble:0.25],
+                    @"UIKeyboardIsLocalUserInfoKey" : [NSNumber numberWithBool:isLocal]
+                  }];
+  shouldIgnore =
+      [viewControllerMock shouldIgnoreKeyboardNotification:notification
+                                          notificationName:UIKeyboardWillChangeFrameNotification];
+  XCTAssertTrue(shouldIgnore == YES);
+
+  // Valid keyboard, keyboard rotated
+  isLocal = YES;
+  notification = [NSNotification
+      notificationWithName:@""
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" :
+                        [NSValue valueWithCGRect:rotatedKeyboardBeginFrame],
+                    @"UIKeyboardFrameEndUserInfoKey" :
+                        [NSValue valueWithCGRect:validKeyboardEndFrame],
+                    @"UIKeyboardAnimationDurationUserInfoKey" : [NSNumber numberWithDouble:0.25],
+                    @"UIKeyboardIsLocalUserInfoKey" : [NSNumber numberWithBool:isLocal]
+                  }];
+  shouldIgnore =
+      [viewControllerMock shouldIgnoreKeyboardNotification:notification
+                                          notificationName:UIKeyboardWillChangeFrameNotification];
+  XCTAssertTrue(shouldIgnore == YES);
+
+  // Valid keyboard, keyboard rotated
+  isLocal = YES;
+  notification = [NSNotification
+      notificationWithName:@""
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameEndUserInfoKey" :
+                        [NSValue valueWithCGRect:validKeyboardEndFrame],
+                    @"UIKeyboardAnimationDurationUserInfoKey" : [NSNumber numberWithDouble:0.25],
+                    @"UIKeyboardIsLocalUserInfoKey" : [NSNumber numberWithBool:isLocal]
+                  }];
+  shouldIgnore =
+      [viewControllerMock shouldIgnoreKeyboardNotification:notification
+                                          notificationName:UIKeyboardWillChangeFrameNotification];
+  XCTAssertTrue(shouldIgnore == NO);
+}
+
+- (void)testCalculateKeyboardAttachMode {
+  FlutterEngine* mockEngine = OCMPartialMock([[FlutterEngine alloc] init]);
+  [mockEngine createShell:@"" libraryURI:@"" initialRoute:nil];
+  FlutterViewController* viewController = [[FlutterViewController alloc] initWithEngine:mockEngine
+                                                                                nibName:nil
+                                                                                 bundle:nil];
+
+  FlutterViewController* viewControllerMock = OCMPartialMock(viewController);
+
+  CGFloat screenWidth = UIScreen.mainScreen.bounds.size.width;
+  CGFloat screenHeight = UIScreen.mainScreen.bounds.size.height;
+
+  OCMStub([viewControllerMock getMainScreen]).andReturn(UIScreen.mainScreen);
+
+  // hide notification
+  CGRect keyboardFrame = CGRectMake(0, 0, 0, 0);
+  NSNumber keyboardMode =
```

Comment:
This doesn't compile:
```suggestion
  NSNumber* keyboardMode =
```
```
../../flutter/shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm:343:12: error: interface type cannot be statically allocated
  NSNumber keyboardMode =
           ^
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8796637329755404225/+/u/build_ios_debug_sim_ios_test_flutter/stdout